### PR TITLE
dao and service layers were created for movie, actor and country models

### PR DESCRIPTION
**Pull request by karma-o**
by using hibernate

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 1]
```diff
@@ -12,11 +16,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return Optional.ofNullable(session.get(Actor.class, id));
+        } catch (HibernateException e) {
+            throw new DataProcessingException("cant get actor by id = " + id, e);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:44]: please fix the exceptions message and in another places too

File: src/main/resources/hibernate.cfg.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        <property name="connection.username">root</property>
+        <property name="connection.password">rootPassword</property>
+        <property name="show_sql">true</property>
+        <property name="hibernate.hbm2ddl.auto">create-drop</property>
+
+        <mapping class="mate.academy.hibernate.relations.model.Actor"/>
+        <mapping class="mate.academy.hibernate.relations.model.Country"/>
+        <mapping class="mate.academy.hibernate.relations.model.Movie"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (liliia-ponomarenko) [inline src/main/resources/hibernate.cfg.xml:15]: don't forget to add an empty line at the end

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java (hunk 1 above)
Comment (karma-o) [reply src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:44]: > please fix the exceptions message and in another places too

I dont see any problem with exceptions, it says about actor as it should be

**Review by liliia-ponomarenko — APPROVED** [review]

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 3]
```diff
@@ -2,11 +2,28 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String title;
-    private List<Actor> actors;
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "movies_actors",
+            joinColumns = @JoinColumn(name = "movie_id", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "actor_id", referencedColumnName = "id"))
+    private List<Actor> actors = new ArrayList<>();
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:26]: Do we need to initialize actors here?

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
 import mate.academy.hibernate.relations.model.Actor;
 import mate.academy.hibernate.relations.service.ActorService;
+import org.hibernate.SessionFactory;
 
 public class ActorServiceImpl implements ActorService {
+    private final ActorDao actorDao;
+
+    public ActorServiceImpl(SessionFactory sessionFactory) {
+        this.actorDao = new ActorDaoImpl(sessionFactory);
+    }
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:14]: Actually better create an object outside of the class and then pass it to the constructor. But if you would like to leave it - remove `this.`, it is redundant

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        this.countryDao = new CountryDaoImpl(sessionFactory);
+    }
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:14]: same here

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

**Review by KostyaNikiforov — APPROVED** [review]

**Pull request closed without merging**